src/error.rs

```diff
@@ -0,0 +1,43 @@
+use std::result::Result as StdResult;
+
+use thiserror::Error as ThisError;
+
+#[allow(missing_docs)]
+pub type Result<T> = StdResult<T, Error>;
```

Comment:
This [already exists](https://github.com/http-rs/surf/blob/3a8a25b40c5f927fb0e425547c5990bcc577fc73/src/lib.rs#L100-L101) from the recent-ish https://github.com/http-rs/surf/commit/ac215cbaa1f612364aff7e1bf5ec282c862f8c90